### PR TITLE
Move daily polkadot compatibilty test from GHA to GitLab

### PR DESCRIPTION
**Pull request by sergejparity**
This PR is created in connection with the https://github.com/paritytech/ci_cd/issues/553
The goal is to reduce precious GitHub actions runtime minutes consumption (saves ~90 minutes per day)

It replaces [Daily compatibility check against latest polkadot](https://github.com/paritytech/staking-miner-v2/actions/workflows/nightly.yml) workflow with the GitLab pipeline of similar functionality.
GitLab pipeline stage `nightly-test` is meant to be executed on the same schedule. Schedule already created and may be activated once this PR will get merged. 

[sample job](https://gitlab.parity.io/parity/mirrors/staking-miner-v2/-/jobs/1718456) - this one without issue creation, because it was added later

[sample failing job](https://gitlab.parity.io/parity/mirrors/staking-miner-v2/-/pipelines/206264) and [sample issue created](https://github.com/paritytech/staking-miner-v2/issues/151)

### CODE REVIEW

**Review by radupopa2010 — APPROVED** [review]

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -47,6 +50,42 @@ variables:
...
+  when:                            on_failure
+  script:
+    - 'curl
+        -X POST
+        -H "Accept: application/vnd.github+json"
+        -H "Authorization: token ${GITHUB_TOKEN}"
+        "https://api.github.com/repos/paritytech/staking-miner-v2/actions/workflows/nightly-fail.yml/dispatches"
+        -d $"{\"ref\":\"main\",\"inputs\":{\"gitlab_pipeline\":\"${CI_PIPELINE_URL}\"}}"'
+  tags:
+    - linux-docker
```
Comment (alvicsam) [inline .gitlab-ci.yml:87]:
```suggestion
    - kubernetes-parity-build
```
Maybe it's better to run it in k8s?

Comment (sergejparity) [reply .gitlab-ci.yml]: Agreed. Also changed an image against thinner `paritytech/tools:latest`

**Review by alvicsam — APPROVED** [review]

**Review by niklasad1 — COMMENTED** [review]
Is it possible to add a link to the status of the Gitlab job in the README as we currently have? 

Not sure if anyone else is using the binary with `--fast-runtime` if so it could built once a day and published somewhere I suppose.

**Comment by sergejparity** [comment]
> Is it possible to add a link to the status of the Gitlab job in the README as we currently have?
> 
> Not sure if anyone else is using the binary with `--fast-runtime` if so it could built once a day and published somewhere I suppose.

Yes. I think its doable. I just need to update `nightly.yml` logic not only to be triggered on test job failure, but every time and make GHA to follow GitLab job's status. So basically the existing link in the README will continue to work.

**Review by niklasad1 — APPROVED** [review]

**Pull request merged**